cpp/src/copying/concatenate.cu

```diff
@@ -336,6 +336,15 @@ std::unique_ptr<column> concatenate(std::vector<column_view> const& columns_to_c
                            [&type](auto const& c) { return c.type() == type; }),
                "Type mismatch in columns to concatenate.");
 
+  if (is_fixed_point(columns_to_concat.front().type())) {
+    CUDF_EXPECTS(std::adjacent_find(columns_to_concat.cbegin(),
```

Comment:
`adjacent_find` is neat, but I think this is clearer and simpler due to the lack of comparison to the end iterator...

```
CUDF_EXPECTS(std::all_of(columns_to_concat.cbegin(), columns_to_concat.cend(),
            [scale = columns_to_concat[0].type().scale()](auto const& c) { return c.type().scale() == scale; }, "...");
```

Reply:
I meant to say I think it's clearer because `all_of` more clearly says what it is expecting.

Reply:
Changed. 

What I really want here it `all_equal`: https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.all_equal